src/base/utils/net.cpp

```diff
@@ -38,5 +38,23 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isLoopbackAddress(const QHostAddress &clientAddress)
+        {
+            return clientAddress == QHostAddress::LocalHost
+                    || clientAddress == QHostAddress::LocalHostIPv6
+                    || clientAddress == QHostAddress("::ffff:127.0.0.1");
+        }
+
+        bool isLANAddress(const QHostAddress &clientAddress)
+        {
+            return clientAddress.isInSubnet(QHostAddress::parseSubnet("10.0.0.0/8"))
```

Comment:
use `isInSubnet(const QHostAddress &subnet, int netmask)`?
i.e. `isInSubnet(QLatin1String("10.0.0.0"), 8)`
and below